changefeedccl: fix PTS test

asg0451 opened the pull request:
Fix failing TestPTSRecordProtectsTargetsAndSystemTables test

Fixes: #135639
Fixes: #138066
Fixes: #137885
Fixes: #137505
Fixes: #136396
Fixes: #135805
Fixes: #135639

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/138243)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: fix PTS test

Fix failing TestPTSRecordProtectsTargetsAndSystemTables test

Fixes: #135639
Fixes: #138066
Fixes: #137885
Fixes: #137505
Fixes: #136396
Fixes: #135805
Fixes: #135639

Release note: None

stevendanna commented on pkg/ccl/changefeedccl/protected_timestamps_test.go:
```diff
@@ -451,29 +453,74 @@ func TestPTSRecordProtectsTargetsAndSystemTables(t *testing.T) {
+	ctx := context.Background()
+
+	// Useful for debugging.
+	require.NoError(t, log.SetVModule("spanconfigstore=2,store=2,reconciler=2,mvcc_gc_queue=2,kvaccessor=2"))
```
Do we still need this?

asg0451 replied:
We don't need it, but if it flakes again i'd like to have them.

stevendanna commented on pkg/ccl/changefeedccl/protected_timestamps_test.go:
```diff
@@ -523,7 +570,18 @@ func TestPTSRecordProtectsTargetsAndSystemTables(t *testing.T) {
 	// Change the user's password to update the users table.
 	sqlDB.Exec(t, `ALTER USER test WITH PASSWORD 'testpass'`)
 
-	time.Sleep(2 * time.Second)
```
I think for this test to be valid , we likely want to sleep some amount of time to ensure that `ts` is beyond the gc threshold before we try to gc the range.

stevendanna commented on pkg/spanconfig/spanconfigreconciler/reconciler.go:
```diff
@@ -461,6 +461,10 @@ func updateSpanConfigRecords(
 			return err // not a retryable error, bubble up
 		}
+
+		if log.V(2) {
+			log.Infof(ctx, "successfully updated span config records: deleted = %+#v; upserted = %+#v", toDelete, toUpsert)
+		}
```
Even at V(2) I wonder if we want this particular log message.   It's rather chatty and hard to read. I leave it to your judgement.

stevendanna reviewed (commented):
Thanks for continuing to dig into this. Apologies for adding this flaky test :D.

I left one more small comment. Don't feel the need to block on my approval though.

Condolences on now being a span configuration expert.

asg0451 commented:
bors r=stevendanna
